recursion/recurse.rb

```diff
@@ -1,10 +1,30 @@
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  end
+  return n * fact(n - 1)
 end
 
 def fib(n)
+  if n == 1
+    return 1
+  elsif n == 2
+    return 1
+  else
+    return fib(n - 1) + fib(n - 2)
+  end
 end
 
 def pal(s)
+  if s.length == 1 || s.length == 0
+    return true
+  elsif s[0] != s[s.length - 1]
+    return false
```

Comment:
YES! You covered all the base cases first and then went to the recursive case. Nice :)